plugins/search-react/src/context/SearchContextForStorybook.stories.tsx

```diff
@@ -18,35 +18,54 @@ import { SearchResultSet } from '@backstage/plugin-search-common';
 import { TestApiRegistry } from '@backstage/test-utils';
 import React, { ComponentProps, PropsWithChildren } from 'react';
 import { searchApiRef } from '../api';
-import { SearchContextProvider as RealSearchContextProvider } from './SearchContext';
+import { SearchContextProvider } from './SearchContext';
 
-type QueryResultProps = {
+/**
+ * Props for {@link SearchApiProviderForStorybook}
+ * @public
+ */
+export type SearchApiProviderForStorybookProps = ComponentProps<
+  typeof SearchContextProvider
+> & {
   mockedResults?: SearchResultSet;
 };
 
 /**
- * Utility context provider only for use in Storybook stories. You should use
- * the real `<SearchContextProvider>` exported by `@backstage/plugin-search-react` in
- * your app instead of this! In some cases (like the search page) it may
- * already be provided on your behalf.
+ * Props for {@link SearchContextProviderForStorybook}
+ * @public
  */
-export const SearchContextProvider = (
-  props: ComponentProps<typeof RealSearchContextProvider> & QueryResultProps,
-) => {
-  return (
-    <SearchApiProvider {...props}>
-      <RealSearchContextProvider children={props.children} />
-    </SearchApiProvider>
-  );
-};
+export type SearchContextProviderForStorybookProps = PropsWithChildren<{
```

Comment:
We'd typically export a mock API implementation instead, which you can use both in tests, dev environments, or storybook. You'd use the `TestApiProvider` to inject it in the storybook.

Already here though so not a blocker for the PR, just ideas for the future